### PR TITLE
feat(PORTALS-982): 

### PR DESCRIPTION
**Pull request by leem42**
- Added outline of access list component, added necessary api calls
- Added type parameter to doPost synapse client function. Added type params to all functions calling doPost. 
- Added type to create entity, returns entity based on entity type passed in
- Updated contributing.md to include section on adding a new component

### CODE REVIEW

File: src/lib/containers/access_requirement_list/AccessRequirementList.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  /**
+   * Returns rendering for the access requirement.
+   *
+   * Only supports SelfSignAccessRequirement and TermsOfUseAccessRequirement
+   *
+   * @param {AccessRequirement} accessRequirement accessRequirement being rendered
+   */
+  const renderAccessRequirement = (accessRequirement: AccessRequirement) => {
+    switch (accessRequirement.concreteType) {
+      case 'org.sagebionetworks.repo.model.SelfSignAccessRequirement':
```
Comment (conundrumgirl) [inline src/lib/containers/access_requirement_list/AccessRequirementList.tsx]: declare those as constants?

File: src/lib/containers/access_requirement_list/SelfSignAccessRequirement.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react'
+import { AccessRequirement } from 'lib/utils/synapseTypes/AccessRequirement/AccessRequirement'
+
+type Props = {
```
Comment (conundrumgirl) [inline src/lib/containers/access_requirement_list/SelfSignAccessRequirement.tsx]: is the generic name 'Props' intentional?

File: src/lib/utils/SynapseClient.ts [hunk 3]
```diff
@@ -1797,6 +1803,78 @@ export const getRestrictionInformation = (
...
+  id: string,
+): Promise<Array<AccessRequirement>> => {
+  let isMoreData = true
+  const accessRequirementResults = [] as AccessRequirement[]
+  const limit = 50
+  let offset = 0
+  while (isMoreData) {
+    const data = await getAccessRequirement(sessionToken, id, limit, offset)
+    accessRequirementResults.concat(data.results)
+    if (data.totalNumberOfResults < 50) {
```
Comment (conundrumgirl) [inline src/lib/utils/SynapseClient.ts]: I don't think  `accessRequirementResults.concat(data.results)` modifies accessRequirementResults. I think it should.

Comment (leem42) [reply src/lib/utils/SynapseClient.ts]: Whoops! Good call, that would've been a fun bug

**Pull request merged**